src/EventTickets/Routes/GetEventFormsListTable.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+namespace Give\EventTickets\Routes;
+
+use Give\DonationForms\Models\DonationForm;
+use Give\EventTickets\Models\Event;
+use Give\EventTickets\Models\EventTicket;
+use Give\Framework\QueryBuilder\QueryBuilder;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+
+/**
+ * @unreleased
+ */
+class GetEventFormsListTable
+{
+    /**
+     * @var string
+     */
+    protected $endpoint = 'events-tickets/event/(?P<event_id>\d+)/forms/list-table';
+
+    /**
+     * @var WP_REST_Request
+     */
+    protected $request;
+
+    /**
+     * @inheritDoc
+     *
+     * @unreleased
+     */
+    public function registerRoute(): void
+    {
+        register_rest_route(
+            'give-api/v2',
+            $this->endpoint,
+            [
+                [
+                    'methods' => WP_REST_Server::READABLE,
+                    'callback' => [$this, 'handleRequest'],
+                    'permission_callback' => [$this, 'permissionsCheck'],
+                ],
+                'args' => [
+                    'event_id' => [
+                        'type' => 'integer',
+                        'sanitize_callback' => 'absint',
+                        'validate_callback' => function ($eventId) {
+                            return Event::find($eventId);
+                        },
+                        'required' => true,
+                    ],
+                    'page' => [
+                        'type' => 'integer',
+                        'default' => 1,
+                        'minimum' => 1
+                    ],
+                    'perPage' => [
+                        'type' => 'integer',
+                        'default' => 30,
+                        'minimum' => 1
+                    ],
+                    'sortColumn' => [
+                        'type' => 'string',
+                        'default' => 'id',
+                        'sanitize_callback' => 'sanitize_text_field',
+                    ],
+                    'sortDirection' => [
+                        'type' => 'string',
+                        'default' => 'asc',
+                        'enum' => ['asc', 'desc'],
+                    ],
+                    'locale' => [
+                        'type' => 'string',
+                        'required' => false,
+                        'default' => get_locale(),
+                    ],
+                ],
+            ]
+        );
+    }
+
+    /**
+     * @unreleased
+     */
+    public function handleRequest(WP_REST_Request $request): WP_REST_Response
+    {
+        $this->request = $request;
```

Comment:
The `WP_REST_API` controllers should not be treated as an individual instance, since the class is instantiated at a global level to register the route. Instance data, such as the current request, should not be hoisted to a class property and instead should be passed into method calls.

If the process is more complicated then it can be extracted into an Action or a special processing action, instead of building up the functionality inside of the controller.